runtime/doc/lsp.txt

```diff
@@ -377,6 +377,8 @@ LSP HIGHLIGHT                                                    *lsp-highlight*
 
 Reference Highlights:
 
+Highlights that are meant to be used when utilizing
+|vim.lsp.buf.document_highlight()|.
```

Comment:
Maybe add a link to the specs (https://microsoft.github.io/language-server-protocol/specification#textDocument_documentHighlight)? I found the difference between the kinds somewhat confusing (especially `Text`).

Reply:
Good point, I added the link.